src/cardContainer.ts

```diff
@@ -27,9 +44,27 @@ export function addCard(card: Card) {
   cardsMap.set(card.name, card)
 }
 
-export function getCard(name: string): CardInstance {
-  return {
-    card: findCard(name),
-    traitPoints: 0
+export function getCard(name: string, metadataArray: Array<string[]>): CardInstance {
+  const card = findCard(name)
+  let metadata: { [key: string]: string } = {}
+  metadataArray.forEach(([key, value]) => {
+    metadata[key] = value
+  })
+
+  const inst: CardInstance = {
+    card: card,
+    traitPoints: 0,
+    finalA: 0,
+    finalB: 0,
+    metadata: metadata
+  }
+  if(card.metadataRequired !== undefined) {
+    card.metadataRequired.forEach(([key, _]) => {
+      if(metadata[key] === undefined) {
+        throw new Error(`missing metadata field ${key}`)
+      }
+    })
   }
```

Comment:
You've got a lot of nested ifs here.
I'd probably write this differently.

```suggestion
  if(card.metadataRequired === undefined) {
    return inst;
  }
 
    card.metadataRequired.forEach(([key, _]) => {
      if(metadata[key] === undefined) {
        throw new Error(`missing metadata field ${key}`)
      }
    })
```